Library/Homebrew/cask_dependent.rb

```diff
@@ -33,11 +33,27 @@ def requirements
       dsl_reqs = @cask.depends_on
 
       dsl_reqs.arch&.each do |arch|
-        requirements << ArchRequirement.new([:x86_64]) if arch[:bits] == 64
-        requirements << ArchRequirement.new([arch[:type]])
+        arch = if arch[:bits] == 64
+          if arch[:type] == :intel
+            :x86_64
+          else
+            :"#{arch[:type]}64"
+          end
+        elsif arch[:type] == :intel && arch[:bits] == 32
+          :i386
+        else
+          arch[:type]
+        end
+        requirements << ArchRequirement.new([arch])
       end
       dsl_reqs.cask.each do |cask_ref|
-        requirements << Requirement.new([{ cask: cask_ref }])
+        requirements << Class.new(Requirement) do
```

Comment:
Could a we declare a new `Requirement` class for this rather than doing it inline?